fvm/src/executor/validator.rs

```diff
@@ -0,0 +1,132 @@
+use anyhow::anyhow;
+use cid::Cid;
+use fvm_ipld_encoding::{Cbor, RawBytes, DAG_CBOR};
+use fvm_shared::error::ExitCode;
+use fvm_shared::message::params::ValidateParams;
+use fvm_shared::message::Message;
+
+use super::{ApplyKind, ApplyRet, DefaultExecutor, Executor, ValidateExecutor, ValidateRet};
+use crate::call_manager::{CallManager, ExecutionType, InvocationResult};
+use crate::kernel::{Block, Context, ExecutionError};
+use crate::machine::Machine;
+use crate::Kernel;
+
+/// TODO try not to be stuck with Default, but it has methods methods i want for validate.
+pub struct DefaultValidateExecutor<K: Kernel>(pub DefaultExecutor<K>);
+
+impl<K> Executor for DefaultValidateExecutor<K>
+where
+    K: Kernel,
+{
+    type Kernel = K;
+
+    /// This is the entrypoint to execute a message.
+    fn execute_message(
+        &mut self,
+        msg: Message,
+        apply_kind: ApplyKind,
+        raw_length: usize,
+    ) -> anyhow::Result<ApplyRet> {
+        self.0.execute_message(msg, apply_kind, raw_length)
+    }
+
+    fn flush(&mut self) -> anyhow::Result<Cid> {
+        self.0.flush()
+    }
+}
+
+impl<K> ValidateExecutor for DefaultValidateExecutor<K>
+where
+    K: Kernel,
+{
+    type Validator = K;
+
+    /// validate a message from an abstract account with a delegate signature
+    fn validate_message(&mut self, msg: Message, sig: Vec<u8>) -> anyhow::Result<ValidateRet> {
+        const VALIDATION_GAS_LIMIT: i64 = i64::MAX; // TODO reasonable gas limit
```

Comment:
Yeah... can you define a constant in fvm_shared? We can set it to `i64::MAX` for now, but we'll need to figure this out later.